modules/r2dbc/src/main/java/org/testcontainers/r2dbc/AbstractR2DBCDatabaseContainerProvider.java

```diff
@@ -0,0 +1,75 @@
+package org.testcontainers.r2dbc;
+
+import io.r2dbc.spi.ConnectionFactoryOptions;
+import lombok.NonNull;
+import org.testcontainers.utility.TestcontainersConfiguration;
+
+import java.util.Properties;
+
+public abstract class AbstractR2DBCDatabaseContainerProvider implements R2DBCDatabaseContainerProvider {
+
+    final String originalDriver;
+
+    protected AbstractR2DBCDatabaseContainerProvider(@NonNull String originalDriver) {
+        this.originalDriver = originalDriver;
+    }
+
+    @Override
+    public boolean supports(ConnectionFactoryOptions options) {
+        return originalDriver.equals(getDriver(options));
+    }
+
+    private String getDriver(ConnectionFactoryOptions options) {
+        Properties properties = TestcontainersConfiguration.getInstance().getProperties();
+        return properties.getProperty(
+            driverPropertyName(options.getRequiredValue(ConnectionFactoryOptions.DRIVER)),
+            originalDriver
```

Comment:
```suggestion
            options.getRequiredValue(ConnectionFactoryOptions.DRIVER)
```

This will fix failing `TestcontainersR2DBCConnectionFactoryTest.failsOnUnknownProvider() `.

Fix is already included in #3019